use-cases/call-center/src/main/java/org/acme/callcenter/service/SolverService.java

```diff
@@ -45,7 +45,7 @@ private void pinCallAssignedToAgents(List<Call> calls) {
     public void startSolving(CallCenter inputProblem,
             Consumer<CallCenter> bestSolutionConsumer, Consumer<Throwable> errorHandler) {
         solverManager.solveAndListen(SINGLETON_ID, id -> inputProblem, bestSolution -> {
-            if (bestSolution.getScore().isSolutionInitialized()) {
+            if (bestSolution.getScore().isSolutionInitialized() && bestSolution.isFeasible()) {
```

Comment:
`Score.isFeasible()` already covers (un)initialized solutions.

```suggestion
            if (bestSolution.isFeasible()) {
```